common/scripted_effects/wc_scripted_effects.txt

```diff
@@ -3690,4 +3690,69 @@ decrease_mind_control_stage_effect = {
 			}
 		}
 	}
+}
+
+support_scarlet_light_effect = {
+	religion = scarlet_light
+	if = {
+		limit = { has_secret_religion = yes }
+		clear_secret_religion = yes
+	}
+	if = {
+		limit = { NOT = { trait = zealous } }
+		if = {
+			limit = { trait = cynical }
+			remove_trait = cynical
+		}
```

Comment:
There is no need for this block, since opposite traits are removed from character on trait adding

Reply:
Refactored!